internal/clients/nightfall/nightfall.go

```diff
@@ -278,3 +284,34 @@ func (n *Client) ReviewDiff(
 
 	return comments, nil
 }
+
+func filterFileDiffs(fileDiffs []*diffreviewer.FileDiff, fileIncludeList, fileExcludeList []string) []*diffreviewer.FileDiff {
+	filteredFileDiffs := fileDiffs
+	if fileIncludeList != nil && len(fileIncludeList) > 0 {
+		filteredFileDiffs = filterByFilePath(filteredFileDiffs, fileIncludeList, true)
+	}
+	if fileExcludeList != nil && len(fileExcludeList) > 0 {
+		filteredFileDiffs = filterByFilePath(filteredFileDiffs, fileExcludeList, false)
+	}
+	return filteredFileDiffs
+}
+
+func filterByFilePath(fileDiffs []*diffreviewer.FileDiff, regexPatterns []string, include bool) []*diffreviewer.FileDiff {
+	filteredFileDiffs := []*diffreviewer.FileDiff{}
+	for _, fd := range fileDiffs {
+		matched := matchRegex(fd.PathNew, regexPatterns)
+		if (matched && include) || (!matched && !include) {
```

Comment:
can we add a comment explaining what these 2 conditions are? I think the logic is sound just a little confusing at first